photon-client/src/main/scala/com/linkedin/photon/ml/util/IOUtils.scala

```diff
@@ -365,4 +367,41 @@ object IOUtils {
 
     builder.mkString
   }
+
+  /**
+   * Save GAME per-group evaluation in terms of (random effect id, evaluation value) to HDFS.
+   *
+   * @param sparkSession The Spark session
+   * @param outputDir The directory in HDFS where to save the evaluation
+   * @param evaluationResultsOpt An Option of evaluation results
+   * @param logger The logger instance for the application
+   */
+  def saveGameEvaluationToHDFS(
+    sparkSession: SparkSession,
+    outputDir: Path,
+    evaluationResultsOpt: Option[EvaluationResults],
+    logger: PhotonLogger): Unit = {
+
+    evaluationResultsOpt match {
+      case Some(evaluationResults) => evaluationResults.evaluations.foreach {
+        case (evaluatorType, (_, Some(groupEval))) => {
+          val evaluatorName = evaluatorType.name.split(MultiEvaluatorType.shardedEvaluatorIdNameSplitter)
+          logger.debug(s"Save per-group evaluation of ${evaluatorName(0)} on ${evaluatorName(1)}")
+          val evalOutputDir = new Path(new Path(outputDir, evaluatorName(0)), evaluatorName(1))
+          sparkSession
+            .createDataFrame(groupEval)
+            .toDF(evaluatorName(1), evaluatorName(0))
+            .coalesce(1)
```

Comment:
not sure you want to do coalesce here, 
1) it is slow
2) if there were 1 million groups, the single file will be large.

Reply:
No, the single file will be small even with 1 million groups. Note that each record is a tuple of (String, Double). I generated an example output which contains 400 records in 2.75KB. So a million records cost roughly ~6.9MB.

Reply:
If not apply coalesce, even for this 2.75KB Avro file, it outputs tens of tiny files.